Container dies when being too specific with a provider

I copied the original config from mr. smurf, but ddns crashes out with:

`--zone-name ok8.sh --record-name ipv4.ok8.sh --provider icanhazip`

```
cloudflare-ddns
version: v1.0.1 (26d548e1b8e67fbd3975a84d812b0e0eb9c2bbfd)

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/larivierec/cloudflare-ddns/pkg/cmd.getProvider(...)
	/go/src/github.com/larivierec/cloudflare-ddns/pkg/cmd/ddns.go:301
github.com/larivierec/cloudflare-ddns/pkg/cmd.initialize()
	/go/src/github.com/larivierec/cloudflare-ddns/pkg/cmd/ddns.go:282 +0x2fa

```

[HelmRelease.yaml](https://github.com/kashalls/home-cluster/blob/main/kubernetes/main/apps/networking/cloudflare-ddns/app/helmrelease.yaml)

--- a/pkg/cmd/ddns.go
+++ b/pkg/cmd/ddns.go
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"os"
 	"os/signal"
+	"strings"
 	"syscall"
 	"time"
 
@@ -112,6 +113,7 @@ func Start(chosenMode string) {
 
 	createProvider()
 	createCloudProvider()
+	initialize()
 
 	if mode != serverless {
 		metrics.InitMetrics()
@@ -195,6 +197,7 @@ func startHttpServer() {
 func StartServerless(w http.ResponseWriter, r *http.Request) {
 	createProvider()
 	createCloudProvider()
+	initialize()
 	zoneName := os.Getenv("ZONE_NAME")
 	recordName := os.Getenv("RECORD_NAME")
 
@@ -252,11 +255,11 @@ func stopServer() {
 }
 
 func createProvider() {
+	ipProviderName = strings.ToLower(strings.TrimSpace(ipProviderName))
 	switch ipProviderName {
 	case "ipify":
 		ipProviders = append(ipProviders, &ipprovider.Ipify{})
-	case "icanhazip":
-	case "icanhaz":
+	case "icanhazip", "icanhaz":
 		ipProviders = append(ipProviders, &ipprovider.ICanHazIp{})
 	case "random":
 		ipProviders = append(ipProviders, &ipprovider.Ipify{})
@@ -275,7 +278,6 @@ func createCloudProvider() {
 	default:
 		cloudProviderObj = cloudflare.NewCloudflareProvider()
 	}
-	initialize()
 }
 
 func initialize() {

